src/electron/go_vpn_tunnel.ts

```diff
@@ -239,6 +241,7 @@ class GoTun2socks {
     args.push('-proxyPort', `${this.config.port}`);
     args.push('-proxyPassword', this.config.password || '');
     args.push('-proxyCipher', this.config.method || '');
+    args.push('-proxyPrefix', encodeURI(this.config.prefix || ''));
```

Comment:
I don't think you need to encode anything.
I think we are spawning the process with a list of arguments, not with a command-line.
Let's remove the encoding if it works.

Reply:
OK, I removed this encoding.  I confirmed that Node uses UTF-8 to encode `child_process` args, so this means that the Go CLI utility has to un-UTF-8 those arguments.  I updated https://github.com/Jigsaw-Code/outline-go-tun2socks/pull/98 with that logic.

Reply:
I don't think we need to care about encodings.
We have two formats for the prefix: string or byte[].
If it's a string, we iterate over the (unicode) characters, using the unicode codepoint as the byte value.
At that point there's no encoding, the String implementation is returning numbers (unicode characters) to us.

Reply:
Please look [here](https://github.com/Jigsaw-Code/outline-go-tun2socks/pull/98/files#diff-af670972d38354924ac59bace6854c98e8379fb42753d4f6be70922fb9b70ae2R108-R124) to see the conversion that is required.

Strings in Javascript, Java, and Swift are (more or less) Unicode natively.  However, the nodejs `child_process.spawn()` functions pass arguments to the child process in UTF-8.  Go strings are byte sequences, not unicode-aware, so we start with a Go string containing UTF-8 bytes.  Then we need to convert from UTF-8 to Unicode codepoints (which Go calls "runes") and extract their numeric values.

I think this is all fine.